Final Project Deliverable 1

mattgilgo opened the pull request:
## Description of Change

Included additional documentation for API sources
Tested data acquisition and updated limitations
Performed preliminary analysis of stock price vs tweet volume

## Team Name and Contributers

George Padavick
Matt GIlgo

## Checklist

Before creating your PR, please ensure the following requirements.

### General

- [x] PR does **not** include irrelevant or hidden files (i.e. `__pycache__`, `.vscode`... etc.)
- [x] Changes are in the correct folder
- [x] There are no merge conflicts
- [x] Commit messages are sensible and descriptive

### Code

- [x] Implementation matches proposed design
- [x] Documentation on how to run the code is created/updated
- [x] Code is legible

### Report

- [x] Report meets the deliverable criteria
- [x] Proposal is up to date with latest decisions

georgep26 committed:
Added a test for alpha vantage using stocks from proposal

georgep26 committed:
Adding .gitignore file

georgep26 committed:
Tested the twitter api

georgep26 committed:
Added items to schedule, deliverables

mattgilgo committed:
reorganized folder structure for planning

georgep26 committed:
Merge pull request #1 from georgep26/yfinance_testing

reorganized folder structure for planning

georgep26 committed:
Tweets count data

georgep26 committed:
Merge pull request #2 from georgep26/twitter_api_testing

Tweets count data

mattgilgo committed:
final edits for deliverable 1

mattgilgo committed:
Merge pull request #3 from georgep26/yfinance_testing

final edits for deliverable 1

georgep26 committed:
Adding twitter testing notebook

georgep26 committed:
Merge pull request #4 from georgep26/twitter_api_testing

Adding twitter testing notebook

mattgilgo committed:
adding stock testing j notebook

mattgilgo committed:
Merge pull request #5 from georgep26/yfinance_testing

adding stock testing j notebook

gallettilance commented:
@mattgilgo @georgep26 Please check that you never committed any key data in the above commits. After a first pass I couldn't find any reference to keys but please double check before I merge this.

georgep26 committed:
Reorganized git repo

georgep26 committed:
adding twitter api script

mattgilgo committed:
editing to remove API reference keys

mattgilgo commented:
> @mattgilgo @georgep26 Please check that you never committed any key data in the above commits. After a first pass I couldn't find any reference to keys but please double check before I merge this.

Lance, thanks for the heads up here! We did find one instance in the code where one key was left in, but we pulled that off and into a local json file. Thank you for the help here, should be all good to go!

gallettilance commented on gilgo_padavick_final_project/deliverables/deliverable_1/README.md:
```diff
@@ -0,0 +1,23 @@
+- [x] Identify key attributes in each data source and limitations 
+- [x] Plot price vs volume of tweets with $ tag
+
+![tweetsplot](tweetsplot.jpg)
```
this path seems wrong. It doesn't display the image in GitHub.

georgep26 replied:
Sorry about that! Just realized the 'jpg' extension got capitalized somehow in the file upload. Should be fixed now.

gallettilance commented on gilgo_padavick_final_project/deliverables/deliverable_1/README.md:
```diff
@@ -0,0 +1,23 @@
+## Project Deliverable 1
```
You need to answer a few key questions - not just plot the data.

mattgilgo replied:
Question 1 of the analysis: Can we classify specific market behavior as a pump?
Answer: The identification of a pump can be easily seen through historical data, with daily charts showing what appears to be a blip in the price when viewing it from the 1Y or 2Y view. To get more specific on how to classify them as they start and not just after the fact, features in the data such as average volume per minute and percentage of price change give us a tip about if the stock is being either pumped or dumped as well. Pairing that in conjunction with an increase in Twitter activity is what we are trying to quantify, and once we are able to work through the Twitter API issues, it will be easier to calculate what percentage changes constitute a "pump" and help our model make clear decisions.

Question 3: What is the impact to new investors, positive or negative, that trade in reaction to these tweets?
Answer: Again, working through the Twitter API will give us an idea about where the masses have peaked in there mentions on twitter about the stock, also telling us where to calculate the price change from when looking at the total impact. In general though, there are stocks such as Kodak, SHIP, and Nikola that all dropped 80%+ from the peaks of their pump prices, so while we know that is a large impact on the potential trader buying at the highest price (AKA the "bagholder"), we aren't able to quantify as a number the impact to the average trader. We will seek to have this answered as we move forward in the project.

gallettilance reviewed (changes requested):
This needs some work to meet the expectations of deliverable 1. Please answer a few more key questions.

mattgilgo committed:
Add files via upload

Adding SCRUM meeting notes from 3/16

georgep26 committed:
Added a script for importing twitter data along with test notebook

georgep26 committed:
Added example parquet file

mattgilgo committed:
Add files via upload

Scrum Notes from 3/22 meeting

mattgilgo committed:
Merge pull request #6 from georgep26/data_aquisition

Data aquisition

mattgilgo committed:
Add files via upload

Uploading helping columns text file

georgep26 committed:
Adding notes from data investigation

georgep26 committed:
Merge branch 'master' of https://github.com/georgep26/CS506Spring2021Repository

georgep26 committed:
Adding grouped DLPN data

mattgilgo committed:
Update README.md

Updates to README.md for Deliverable 2. Still need an image of the refined preliminary analysis as well as a new answered question from Deliverable 1.

mattgilgo committed:
Update README.md

Updating with Pump and Dump stock examples 2020

mattgilgo committed:
Add files via upload

Updated Stock_Testing Notebook

georgep26 committed:
Added changes for deliverable 2

georgep26 committed:
Resolving conflicts

georgep26 committed:
Adding documentation

georgep26 committed:
Update README.md

mattgilgo commented:
Changed requested resolved
